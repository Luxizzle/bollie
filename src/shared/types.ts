export interface Token {
  access_token: string;
  token_type: string;
  expires_in: number;
  expires_at: number;
  scope: string;
}
