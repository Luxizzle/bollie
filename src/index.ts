export * from './shared/index'
export * as v5 from './v5/index';
