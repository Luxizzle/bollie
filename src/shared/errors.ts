export class RateLimitError extends Error {}

export class RateLimitMaxRetriesError extends Error {}
