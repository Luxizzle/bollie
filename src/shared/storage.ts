import { Token } from './types';
import { readFile, writeFile } from 'fs-extra';
import path from 'path';
import { mkdirSync } from 'fs';

export interface IStorage {
  getToken(): PromiseLike<Token | undefined>;
  setToken(token: Token): PromiseLike<void>;
}

export class MemoryStorage implements IStorage {
  token?: Token;

  async getToken() {
    return this.token;
  }

  async setToken(token: Token) {
    this.token = token;
  }
}

export class FileStorage implements IStorage {
  constructor(protected path: string) {
    mkdirSync(this.path, { recursive: true });
  }

  async getToken() {
    try {
      const token = await readFile(path.join(this.path, 'token.json'), 'utf-8');

      return JSON.parse(token);
    } catch (err) {
      // @ts-ignore
      if (err.code === 'ENOENT') {
        return undefined;
      }
      console.dir(err);

      throw err;
    }
  }

  async setToken(token: Token) {
    await writeFile(
      path.join(this.path, 'token.json'),
      JSON.stringify(token),
      'utf-8'
    );
  }
}
