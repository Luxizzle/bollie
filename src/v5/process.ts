export enum ApiProcessStatus {
  Pending = 'PENDING',
  Success = 'SUCCESS',
  Failure = 'FAILURE',
  Timeout = 'TIMEOUT',
}

export interface ApiProcess {
  processStatusId: string;
  entityId: string;
  eventType: string;
  description: string;
  status: ApiProcessStatus;
  errorMessage: string;
  createTimestamp: string;
  links: [
    {
      rel: 'self';
      href: string;
      method: 'GET';
    }
  ];
}
