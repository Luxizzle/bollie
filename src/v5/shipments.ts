import {
  enums,
  intersection,
  object,
  optional,
  string,
  tuple,
  type,
  union,
} from 'superstruct';

export enum TransporterCode {
  Briefpost = 'BRIEFPOST',
  Ups = 'UPS',
  Tnt = 'TNT',
  TntEXTRA = 'TNT-EXTRA',
  TntBrief = 'TNT_BRIEF',
  TntExpress = 'TNT-EXPRESS',
  Dyl = 'DYL',
  DpdNL = 'DPD-NL',
  DpdBE = 'DPD-BE',
  BpostBe = 'BPOST_BE',
  BpostBrief = 'BPOST_BRIEF',
  Dhlforyou = 'DHLFORYOU',
  Gls = 'GLS',
  FedexNl = 'FEDEX_NL',
  FedexBe = 'FEDEX_BE',
  Other = 'OTHER',
  Dhl = 'DHL',
  DhlDe = 'DHL_DE',
  DhlGlobalMail = 'DHL-GLOBAL-MAIL',
  Tsn = 'TSN',
  Fiege = 'FIEGE',
  Transmission = 'TRANSMISSION',
  ParcelNl = 'PARCEL-NL',
  Logoix = 'LOGOIX',
  Packs = 'PACKS',
  Courier = 'COURIER',
  Rjp = 'RJP',
}

export type ApiPutShipmentBody = {
  orderItems: [{ orderItemId: string }];
  shipmentReference?: string;
} & (
  | { shippingLabelId: string }
  | {
      transport: {
        trackAndTrace: string;
        transporterCode: TransporterCode;
      };
    }
);

export const ShipmentBodySchema = intersection([
  union([
    type({
      shippingLabelId: string(),
    }),
    type({
      transport: object({
        trackAndTrace: string(),
        transporterCode: enums(Object.values(TransporterCode)),
      }),
    }),
  ]),
  type({
    orderItems: tuple([object({ orderItemId: string() })]),
    shipmentReference: optional(string()),
  }),
]);
