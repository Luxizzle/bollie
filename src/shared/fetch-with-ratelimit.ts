import fetch from 'isomorphic-fetch';
import debug from 'debug';
import { RateLimitMaxRetriesError } from './errors';

const log = debug('bollie:fetch');

const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));

export type RateLimitOptions = {
  maxRetries?: number;
  retryAfterDefault?: number;
};

export type RequestInitWithRateLimitOptions = {
  rateLimit?: RateLimitOptions;
} & RequestInit;

export const fetchWithRatelimit = async (
  input: RequestInfo,
  init: RequestInitWithRateLimitOptions
) => {
  const { maxRetries = 10, retryAfterDefault = 10 }: RateLimitOptions =
    init?.rateLimit ?? {};

  let attempted = 0;

  while (attempted <= maxRetries) {
    attempted += 1;

    log('Attempting to fetch %s, attempt %d', input, attempted);
    const response = await fetch(input, init);

    if (response.status === 429) {
      const retryAfterHeader = response.headers.get('Retry-After');
      const retryAfter =
        (retryAfterHeader
          ? parseInt(retryAfterHeader, 10)
          : retryAfterDefault) * 1000;

      log(
        'Got status 429 while fetching %s, retrying after %d seconds',
        input,
        retryAfter
      );

      await sleep(retryAfter);
    } else {
      return response;
    }
  }

  throw new RateLimitMaxRetriesError();
};
