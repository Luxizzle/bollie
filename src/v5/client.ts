import { IStorage, MemoryStorage } from '../shared/storage';
import { SetRequired } from 'type-fest';
import fetch from 'isomorphic-fetch';
import { Token } from '../shared/types';
import { ApiGetOrdersQuery, ApiOrder, ApiReducedOrder } from './order';
import { fetchWithRatelimit } from '../shared/fetch-with-ratelimit';
import debug from 'debug';

const log = debug('bollie:client');

export interface ClientOptions {
  clientId: string;
  clientSecret: string;

  tokenEndpoint?: string;
  apiEndpoint?: string;
  demo?: boolean;

  storage?: IStorage;
}

export class Client {
  protected readonly options: SetRequired<
    ClientOptions,
    'tokenEndpoint' | 'apiEndpoint'
  >;

  protected readonly storage: IStorage;

  constructor(options: ClientOptions) {
    log('Creating new client');
    this.options = {
      tokenEndpoint: 'https://login.bol.com',
      apiEndpoint: 'https://api.bol.com',
      demo: false,
      ...options,
    };

    this.storage = this.options.storage ?? new MemoryStorage();
  }

  protected get endpoint() {
    return (
      this.options.apiEndpoint +
      (this.options.demo ? '/retailer-demo' : '/retailer')
    );
  }

  async getFetchOptions(): Promise<RequestInit> {
    const token = await this.getToken();

    return {
      headers: {
        Accept: 'application/vnd.retailer.v5+json',
        'Content-Type': 'application/vnd.retailer.v5+json',
        Authorization: `${token.token_type} ${token.access_token}`,
      },
    };
  }

  async getToken(force: boolean = false): Promise<Token> {
    let token = await this.storage.getToken();

    if (!token || (token && token.expires_at < Date.now()) || force) {
      log('Token expired or unavailable in storage. Fetching new token.');

      token = await this.fetchToken();

      await this.storage.setToken(token);
    }

    return token;
  }

  protected fetchToken(): Promise<Token> {
    return fetch(
      `${this.options.tokenEndpoint}/token?grant_type=client_credentials`,
      {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          Authorization: `Basic ${Buffer.from(
            `${this.options.clientId}:${this.options.clientSecret}`
          ).toString('base64')}`,
        },
      }
    )
      .then(response => response.json())
      .then(token => {
        token.expires_at = Date.now() + token.expires_in * 1000;

        return token;
      });
  }

  async getOrders(options: ApiGetOrdersQuery = {}): Promise<ApiReducedOrder[]> {
    log('Getting orders');

    const query = new URLSearchParams();

    if (typeof options.page === 'number') {
      query.set('page', options.page.toString());
    }
    if (typeof options.fulfilmentMethod === 'string') {
      query.set('fulfilment-method', options.fulfilmentMethod);
    }
    if (typeof options.status === 'string') {
      query.set('status', options.status);
    }

    const response = await fetchWithRatelimit(
      `${this.endpoint}/orders?${query.toString()}`,
      {
        ...(await this.getFetchOptions()),
      }
    );

    const body = await response.json();

    const orders: ApiReducedOrder[] = body.orders ?? [];

    return orders;
  }

  async getOrder(orderId: string): Promise<ApiOrder> {
    log('Getting order %s', orderId);

    const response = await fetchWithRatelimit(
      `${this.endpoint}/orders/${orderId}`,
      {
        ...(await this.getFetchOptions()),
      }
    );

    return await response.json();
  }
}
