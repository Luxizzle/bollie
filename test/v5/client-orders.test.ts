import { FileStorage } from '../../src/shared/storage';
import { Client } from '../../src/v5';

const storage = new FileStorage('./test/.tmp/v5');

describe('should be able to fetch orders', () => {
  const client = new Client({
    clientId: process.env.BOL_CLIENT_ID as string,
    clientSecret: process.env.BOL_CLIENT_SECRET as string,
    storage,
    demo: true,
  });

  test('with default options', async () => {
    const orders = await client.getOrders();

    expect(orders).toBeInstanceOf(Array);
    expect(orders).toMatchSnapshot();
  });

  // Not on demo mode, but we can test outside of demo mode, except i dont want
  // these in committed snapshots.

  // test('with different page', async () => {
  //   const client = new Client({
  //     clientId: process.env.BOL_CLIENT_ID as string,
  //     clientSecret: process.env.BOL_CLIENT_SECRET as string,
  //     storage,
  //     // demo:true,
  //   });
  //   const orders = await client.getOrders({ page: 2 });

  //   expect(orders).toBeInstanceOf(Array);
  //   expect(orders).toMatchSnapshot();
  // });

  test('with different fullfillment method', async () => {
    const orders = await client.getOrders({ fulfilmentMethod: 'FBB' });

    expect(orders).toBeInstanceOf(Array);
    expect(orders).toMatchSnapshot();
  });

  test('with different status', async () => {
    const orders = await client.getOrders({ status: 'ALL' });

    expect(orders).toBeInstanceOf(Array);
    expect(orders).toMatchSnapshot();
  });
});

describe('Should be able to get a single order', () => {
  const client = new Client({
    clientId: process.env.BOL_CLIENT_ID as string,
    clientSecret: process.env.BOL_CLIENT_SECRET as string,
    storage,
    demo: true,
  });

  test.each(['1043946570', '1042831430', 'A4K8290LP0', 'B3K8290LP0'])(
    'Demo order %s',
    async orderId => {
      const order = await client.getOrder(orderId);

      expect(order).toMatchSnapshot();
    }
  );
});
