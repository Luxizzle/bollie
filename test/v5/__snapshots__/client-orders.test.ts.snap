// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to get a single order Demo order 1042831430 1`] = `
Object {
  "billingDetails": Object {
    "city": "PLATOONDORP",
    "company": "Pieter Post",
    "countryCode": "NL",
    "email": "2yldzdi2wjcf5ir4sycq7lufqpytxy@verkopen.test2.bol.com",
    "extraAddressInformation": "Extra informatie",
    "firstName": "Pieter",
    "houseNumber": "21",
    "houseNumberExtension": "X",
    "kvkNumber": "99887766",
    "orderReference": "Mijn order ref",
    "salutation": "MALE",
    "streetName": "Skywalkerstraat",
    "surname": "Post",
    "vatNumber": "NL123456789B01",
    "zipCode": "1234AB",
  },
  "orderId": "1042831430",
  "orderItems": Array [
    Object {
      "cancellationRequest": false,
      "commission": 2.22,
      "fulfilment": Object {
        "distributionParty": "RETAILER",
        "expiryDate": "2018-04-21",
        "latestDeliveryDate": "2018-04-18",
        "method": "FBR",
        "timeFrameType": "REGULAR",
      },
      "offer": Object {
        "offerId": "8f6283e3-de98-c92f-e053-3598790a63b5",
        "reference": "MijnOffer0021",
      },
      "orderItemId": "6107331382",
      "product": Object {
        "ean": "8712626055143",
        "title": "Star Wars - The happy family 2",
      },
      "quantity": 1,
      "quantityCancelled": 0,
      "quantityShipped": 1,
      "unitPrice": 22.98,
    },
    Object {
      "cancellationRequest": false,
      "commission": 2.22,
      "fulfilment": Object {
        "distributionParty": "RETAILER",
        "expiryDate": "2018-04-21",
        "latestDeliveryDate": "2018-04-21",
        "method": "FBR",
        "timeFrameType": "REGULAR",
      },
      "offer": Object {
        "offerId": "8f6125e3-de98-c08f-e053-3523590a63b9",
        "reference": "MijnOffer1121",
      },
      "orderItemId": "6107331383",
      "product": Object {
        "ean": "8717418510749",
        "title": "Star Wars - The happy family 2",
      },
      "quantity": 3,
      "quantityCancelled": 0,
      "quantityShipped": 2,
      "unitPrice": 10.3,
    },
  ],
  "orderPlacedDateTime": "2019-04-20T12:58:39+02:00",
  "pickupPoint": false,
  "shipmentDetails": Object {
    "city": "BRUSSEL",
    "countryCode": "BE",
    "email": "2mqmu3hefoawq3mqgzxh4hbpkyh2rs@verkopen.test2.bol.com",
    "extraAddressInformation": "Apartment",
    "firstName": "Hans",
    "houseNumber": "199",
    "houseNumberExtension": "B",
    "language": "nl",
    "salutation": "MALE",
    "streetName": "Skywalkerstraat",
    "surname": "de Grote",
    "zipCode": "1200",
  },
}
`;

exports[`Should be able to get a single order Demo order 1043946570 1`] = `
Object {
  "billingDetails": Object {
    "city": "PLATOONDORP",
    "company": "Pieter Post",
    "countryCode": "NL",
    "email": "2yldzdi2wjcf5ir4sycq7lufqpytxy@verkopen.test2.bol.com",
    "extraAddressInformation": "Extra informatie",
    "firstName": "Pieter",
    "houseNumber": "21",
    "houseNumberExtension": "X",
    "kvkNumber": "99887766",
    "orderReference": "Mijn order ref",
    "salutation": "MALE",
    "streetName": "Skywalkerstraat",
    "surname": "Post",
    "vatNumber": "NL123456789B01",
    "zipCode": "1234AB",
  },
  "orderId": "1043946570",
  "orderItems": Array [
    Object {
      "cancellationRequest": false,
      "commission": 2.22,
      "fulfilment": Object {
        "distributionParty": "RETAILER",
        "expiryDate": "2019-05-04",
        "latestDeliveryDate": "2019-05-01",
        "method": "FBR",
        "timeFrameType": "REGULAR",
      },
      "offer": Object {
        "offerId": "8f6285e3-de98-c97f-e053-3544090a63b3",
        "reference": "MijnOffer6627",
      },
      "orderItemId": "6042823871",
      "product": Object {
        "ean": "8717418510749",
        "title": "Star Wars - The happy family 2",
      },
      "quantity": 3,
      "quantityCancelled": 0,
      "quantityShipped": 3,
      "unitPrice": 11,
    },
  ],
  "orderPlacedDateTime": "2019-04-29T18:18:21+02:00",
  "pickupPoint": true,
  "shipmentDetails": Object {
    "city": "PLATOONDORP",
    "countryCode": "NL",
    "email": "2puwjjic7wxsbmkj3bc2v6jsj2wimk@verkopen.test2.bol.com",
    "firstName": "Pieter",
    "houseNumber": "21",
    "language": "nl",
    "pickupPointName": "Albert Heijn: UTRECHT",
    "salutation": "MALE",
    "streetName": "Skywalkerstraat",
    "surname": "Post",
    "zipCode": "1234AB",
  },
}
`;

exports[`Should be able to get a single order Demo order A4K8290LP0 1`] = `
Object {
  "billingDetails": Object {
    "city": "PLATOONDORP",
    "company": "Pieter Post",
    "countryCode": "NL",
    "email": "2yldzdi2wjcf5ir4sycq7lufqpytxy@verkopen.test2.bol.com",
    "extraAddressInformation": "Extra informatie",
    "firstName": "Pieter",
    "houseNumber": "21",
    "houseNumberExtension": "X",
    "kvkNumber": "99887766",
    "orderReference": "Mijn order ref",
    "salutation": "MALE",
    "streetName": "Skywalkerstraat",
    "surname": "Post",
    "vatNumber": "NL123456789B01",
    "zipCode": "1234AB",
  },
  "orderId": "A4K8290LP0",
  "orderItems": Array [
    Object {
      "cancellationRequest": false,
      "commission": 2.22,
      "fulfilment": Object {
        "distributionParty": "RETAILER",
        "expiryDate": "2019-12-18",
        "latestDeliveryDate": "2019-12-15",
        "method": "FBR",
        "timeFrameType": "REGULAR",
      },
      "offer": Object {
        "offerId": "8f6283e3-de98-c92f-e053-3598790a63b5",
        "reference": "MijnOffer0021",
      },
      "orderItemId": "2070906705",
      "product": Object {
        "ean": "8718846038683",
        "title": "adidas Originals Booklet case Bohemian color for iPhone 7/8 colourful",
      },
      "quantity": 1,
      "quantityCancelled": 0,
      "quantityShipped": 1,
      "unitPrice": 34.99,
    },
  ],
  "orderPlacedDateTime": "2019-12-06T13:04:34+01:00",
  "pickupPoint": false,
  "shipmentDetails": Object {
    "city": "PLATOONDORP",
    "countryCode": "NL",
    "email": "2mqmu3hefoawq3mqgzxh4hbpkyh2rs@verkopen.test2.bol.com",
    "firstName": "Hans",
    "houseNumber": "199",
    "language": "nl",
    "salutation": "FEMALE",
    "streetName": "Skywalkerstraat",
    "surname": "de Grote",
    "zipCode": "1234AB",
  },
}
`;

exports[`Should be able to get a single order Demo order B3K8290LP0 1`] = `
Object {
  "billingDetails": Object {
    "city": "PLATOONDORP",
    "company": "Pieter Post",
    "countryCode": "NL",
    "email": "2yldzdi2wjcf5ir4sycq7lufqpytxy@verkopen.test2.bol.com",
    "extraAddressInformation": "Extra informatie",
    "firstName": "Pieter",
    "houseNumber": "21",
    "houseNumberExtension": "X",
    "kvkNumber": "99887766",
    "orderReference": "Mijn order ref",
    "salutation": "MALE",
    "streetName": "Skywalkerstraat",
    "surname": "Post",
    "vatNumber": "NL123456789B01",
    "zipCode": "1234AB",
  },
  "orderId": "B3K8290LP0",
  "orderItems": Array [
    Object {
      "cancellationRequest": false,
      "commission": 2.22,
      "fulfilment": Object {
        "distributionParty": "RETAILER",
        "exactDeliveryDate": "2020-01-08",
        "expiryDate": "2020-01-09",
        "method": "FBR",
        "timeFrameType": "REGULAR",
      },
      "offer": Object {
        "offerId": "8f6283e3-de98-c92f-e053-3598790a63b5",
        "reference": "MijnOffer0021",
      },
      "orderItemId": "2070906706",
      "product": Object {
        "ean": "8718846038683",
        "title": "adidas Originals Booklet case Bohemian color for iPhone 7/8 colourful",
      },
      "quantity": 1,
      "quantityCancelled": 0,
      "quantityShipped": 1,
      "unitPrice": 34.99,
    },
  ],
  "orderPlacedDateTime": "2020-01-01T13:04:34+01:00",
  "pickupPoint": false,
  "shipmentDetails": Object {
    "city": "PLATOONDORP",
    "countryCode": "NL",
    "email": "2mqmu3hefoawq3mqgzxh4hbpkyh2rs@verkopen.test2.bol.com",
    "firstName": "Hans",
    "houseNumber": "199",
    "language": "nl",
    "salutation": "FEMALE",
    "streetName": "Skywalkerstraat",
    "surname": "de Grote",
    "zipCode": "1234AB",
  },
}
`;

exports[`should be able to fetch orders with default options 1`] = `
Array [
  Object {
    "orderId": "1043946570",
    "orderItems": Array [
      Object {
        "ean": "8717418510749",
        "orderItemId": "6042823871",
        "quantity": 3,
        "quantityCancelled": 0,
        "quantityShipped": 3,
      },
    ],
    "orderPlacedDateTime": "2019-04-29T18:18:21+02:00",
  },
  Object {
    "orderId": "1042831430",
    "orderItems": Array [
      Object {
        "ean": "8712626055143",
        "orderItemId": "6107331382",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
      Object {
        "ean": "8804269223123",
        "orderItemId": "6107331383",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2019-04-20T12:58:39+02:00",
  },
  Object {
    "orderId": "A4K8290LP0",
    "orderItems": Array [
      Object {
        "ean": "8718846038683",
        "orderItemId": "2070906705",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2019-12-06T13:04:34+01:00",
  },
  Object {
    "orderId": "B3K8290LP0",
    "orderItems": Array [
      Object {
        "ean": "8718846038683",
        "orderItemId": "2070906706",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2020-01-01T13:04:34+01:00",
  },
]
`;

exports[`should be able to fetch orders with different fullfillment method 1`] = `
Array [
  Object {
    "orderId": "1043965710",
    "orderItems": Array [
      Object {
        "ean": "8717418510749",
        "orderItemId": "6107989317",
        "quantity": 2,
        "quantityCancelled": 0,
        "quantityShipped": 2,
      },
    ],
    "orderPlacedDateTime": "2019-04-30T21:56:39+02:00",
  },
  Object {
    "orderId": "1042823870",
    "orderItems": Array [
      Object {
        "ean": "8785056370398",
        "orderItemId": "6107771545",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
      Object {
        "ean": "8785073140359",
        "orderItemId": "6107771546",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2019-04-30T16:47:32+02:00",
  },
]
`;

exports[`should be able to fetch orders with different page 1`] = `Array []`;

exports[`should be able to fetch orders with different status 1`] = `
Array [
  Object {
    "orderId": "1043946570",
    "orderItems": Array [
      Object {
        "ean": "8717418510749",
        "orderItemId": "6042823871",
        "quantity": 3,
        "quantityCancelled": 0,
        "quantityShipped": 3,
      },
    ],
    "orderPlacedDateTime": "2019-04-29T18:18:21+02:00",
  },
  Object {
    "orderId": "1042831430",
    "orderItems": Array [
      Object {
        "ean": "8712626055143",
        "orderItemId": "6107331382",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
      Object {
        "ean": "8804269223123",
        "orderItemId": "6107331383",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2019-04-20T12:58:39+02:00",
  },
  Object {
    "orderId": "A4K8290LP0",
    "orderItems": Array [
      Object {
        "ean": "8718846038683",
        "orderItemId": "2070906705",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2019-12-06T13:04:34+01:00",
  },
  Object {
    "orderId": "B3K8290LP0",
    "orderItems": Array [
      Object {
        "ean": "8718846038683",
        "orderItemId": "2070906706",
        "quantity": 1,
        "quantityCancelled": 0,
        "quantityShipped": 1,
      },
    ],
    "orderPlacedDateTime": "2020-01-01T13:04:34+01:00",
  },
]
`;
